src/Compilers/CSharp/Test/Emit3/Semantics/ExtensionTests.cs

```diff
@@ -48432,6 +48995,61 @@ static class E
             Diagnostic(ErrorCode.WRN_MissingTypeParamTag, "T").WithArguments("T", "E.extension<T>(T)").WithLocation(5, 15));
     }
 
+    [Fact]
+    public void XmlDoc_TypeParam_06()
+    {
+        // type parameter documented in different parts
+        var src = """
+static class E
+{
+    /// <summary>Summary for extension block</summary>
+    /// <typeparam name="T">Description for T</typeparam>
+    extension<T, U>(C<T, U> c)
+    {
+    }
+
+    /// <typeparam name="U">Description for U</typeparam>
+    extension<T, U>(C<T, U> c)
+    {
+    }
+}
+
+class C<T, U> { }
+""";
+        var comp = CreateCompilation(src, parseOptions: TestOptions.RegularPreviewWithDocumentationComments);
+        comp.VerifyEmitDiagnostics();
+    }
+
+    [Fact]
+    public void XmlDoc_TypeParam_07()
+    {
+        var src = """
+static class E
+{
+    /// <summary>Summary for extension block</summary>
+    /// <typeparam name="T1">Description for T1</typeparam>
+    extension<T1, U>(C<T1, U> c)
+    {
+    }
+
+    /// <typeparam name="T2">Description for T2</typeparam>
+    extension<T2, U>(C<T2, U> c)
```

Comment:
>T2 [](http://example.com/codeflow?start=14&length=2)

Do we test scenario where this type parameter ins named "T1" (to force merging with the previous extension)?

Reply:
`XmlDoc_05`. Note: it was fixed by tracking type parameters by name (it now reports a warning for documenting type parameter twice)